domain/camera/src/main/java/com/google/jetpackcamera/domain/camera/CameraXCameraUseCase.kt

```diff
@@ -228,15 +246,58 @@ constructor(
         }
     }
 
-    override fun setFlashMode(flashMode: FlashMode) {
+    override fun getScreenFlashEvents() = screenFlashEvents.asSharedFlow()
+
+    override fun setFlashMode(flashMode: FlashMode, isFrontFacing: Boolean) {
+        val isScreenFlashRequired = isScreenFlash(flashMode, isFrontFacing)
+
+        if (isScreenFlashRequired) {
+            imageCaptureUseCase.screenFlashUiControl = object : ScreenFlashUiControl {
+                override fun applyScreenFlashUi(
+                    screenFlashUiCompleter: ImageCapture.ScreenFlashUiCompleter
+                ) {
+                    Log.d(TAG, "applyScreenFlashUi")
+                    coroutineScope.launch {
+                        screenFlashEvents.emit(
+                            CameraUseCase.ScreenFlashEvent(Type.APPLY_UI) {
+                                screenFlashUiCompleter.complete()
```

Comment:
Are we able to cancel this completer? What happens in that case?

The reason I'm asking is we should be able to handle the case where the UI can't keep up with screen flash events, for whatever reason (perhaps the user navigated away from the screen, or there is simply too much going on on the UI thread). We should be able to timeout the screen flash request, or handle the case of buffer overflow in the `MutableSharedFlow`.